build/debs/build.go

```diff
@@ -420,127 +450,173 @@ func main() {
 					Revision:     revision,
-					Distribution: DistributionUnstable,
+					Distribution: DistributionTesting,
 				},
 				{
-					GetVersion:   getCRIToolsLatestVersion,
 					Revision:     revision,
-					Distribution: DistributionTesting,
+					Distribution: DistributionUnstable,
 				},
 			},
 		},
 	}
 
 	if kubeVersion != "" {
-		getSpecifiedVersion := func() (string, error) {
-			return kubeVersion, nil
-		}
 		builds = []build{
 			{
 				Package: "kubectl",
-				Versions: []version{
+				Definitions: []packageDefinition{
 					{
-						GetVersion:          getSpecifiedVersion,
-						Revision:            revision,
-						Distribution:        DistributionStable,
-						GetDownloadLinkBase: getReleaseDownloadLinkBase,
+						KubernetesVersion: kubeVersion,
+						Revision:          revision,
+						Distribution:      DistributionStable,
 					},
 				},
 			},
 			{
 				Package: "kubelet",
-				Versions: []version{
+				Definitions: []packageDefinition{
 					{
-						GetVersion:          getSpecifiedVersion,
-						Revision:            revision,
-						Distribution:        DistributionStable,
-						GetDownloadLinkBase: getReleaseDownloadLinkBase,
+						KubernetesVersion: kubeVersion,
+						Revision:          revision,
+						Distribution:      DistributionStable,
 					},
 				},
 			},
 			{
 				Package: "kubernetes-cni",
-				Versions: []version{
+				Definitions: []packageDefinition{
 					{
-						Version:      cniVersion,
-						Revision:     revision,
-						Distribution: DistributionStable,
+						Version:           cniVersion,
+						Revision:          revision,
+						KubernetesVersion: kubeVersion,
+						Distribution:      DistributionStable,
 					},
 				},
 			},
 			{
 				Package: "kubeadm",
-				Versions: []version{
+				Definitions: []packageDefinition{
 					{
-						GetVersion:          getSpecifiedVersion,
-						Revision:            revision,
-						Distribution:        DistributionStable,
-						GetDownloadLinkBase: getReleaseDownloadLinkBase,
+						KubernetesVersion: kubeVersion,
+						Revision:          revision,
+						Distribution:      DistributionStable,
 					},
 				},
 			},
 			{
 				Package: "cri-tools",
-				Versions: []version{
+				Definitions: []packageDefinition{
 					{
-						GetVersion:   getCRIToolsLatestVersion,
-						Revision:     revision,
-						Distribution: DistributionStable,
+						KubernetesVersion: kubeVersion,
+						Revision:          revision,
+						Distribution:      DistributionStable,
 					},
 				},
 			},
 		}
 	}
 
-	if err := walkBuilds(builds, func(pkg, arch string, v version) error {
+	if err := walkBuilds(builds, func(pkg, arch string, packageDef packageDefinition) error {
 		c := cfg{
-			Package: pkg,
-			version: v,
-			Arch:    arch,
+			packageDefinition: &packageDef,
+			Package:           pkg,
+			Arch:              arch,
 		}
+
+		c.Name = pkg
+
+		var err error
+
+		// TODO: Allow building packages for a specific distro type
```

Comment:
By "distro type" here do you mean OS distribution, a variation for which you've mostly removed support, or the unstable/testing/stable distribution channel?  "distro" is overloaded, so it might be beneficial to use something more evocative of the release distribution channel concept.

Personally I'm ok with us not building distinctly for a dozen different Debian/Ubuntu targets, specifying the one we do build on and that runtime mileage may vary if the host is different than that.  We can't build and test them all.